index.bs

```diff
@@ -3830,6 +3859,802 @@ opened</dfn> steps given |window|, |type| and |message|.
 
 </div>
 
+## The network Module ## {#module-network}
+
+The <dfn export for=modules>network</dfn> module contains commands and events
+relating to network requests.
+
+### Definition ### {#module-network-definition}
+
+<pre class="cddl remote-cddl">
+
+NetworkCommand = (
+)
+
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+NetworkResult = (
+)
+
+NetworkEvent = (
+    network.BeforeRequestSent //
+    network.ResponseStarted //
+    network.ResponseCompleted //
+    network.FetchError
+)
+
+</pre>
+
+A [=remote end=] has a <dfn>before request sent map</dfn> which is initially an
+empty map.  It's used to track the network events for which a
+<code>network.beforeRequestSent</code> event has already been sent.
+
+### Types ### {#module-network-types}
+
+#### The network.BaseParameters type #### {#type-network-BaseParameters}
+
+<pre class="cddl local-cddl">
+network.BaseParameters = {
+    request: network.RequestData,
+    navigation: Navigation / null,
+    context: BrowsingContext / null,
+    timestamp: int,
+    isRedirect: bool,
```

Comment:
Could we try to keep the alphabetical order of fields for each and every CDDL type you added in this PR? That would keep us in sync with already existing ones.

```suggestion
    context: BrowsingContext / null,
    isRedirect: bool,
    navigation: Navigation / null,
    request: network.RequestData,
    timestamp: int,
```

Reply:
We can, but there's not loads of precedent here. The current spec is less consistent than you think. CDP always puts optional fields at the end but otherwise seems to use "importance" order (maybe the non-JSON serializations want new fields added after existing ones for compatibility reasons?). Other web specs like DOM group IDL blocks by function rather than in strictly alphabetical order.